scripts/ci/verify_linter.py

```diff
@@ -0,0 +1,208 @@
+        ---------------------------------------------------------------------------------------------------------
+        """
+        return msg
+
+
+class AzExtension:
+    def __init__(self, extension_name):
+        self.extension_name = extension_name
+
+    @staticmethod
+    def _cmd(cmd):
+        print(cmd)
+        check_call(cmd.split(), shell=True)
+
+    def add_from_url(self, url):
+        self._cmd('az extension add -s {} -y'.format(url))
+
+    def remove(self):
+        self._cmd('az extension remove -n {}'.format(self.extension_name))
+
+
+class AzdevExtension:
+    def __init__(self, extension_name):
+        self.extension_name = extension_name
+
+    @staticmethod
+    def _cmd(cmd):
+        print(cmd)
+        check_call(cmd, shell=True)
+
+    def add_from_code(self):
+        self._cmd('azdev extension add {}'.format(self.extension_name))
+
+    def remove(self):
+        self._cmd('azdev extension remove {}'.format(self.extension_name))
+
+    def linter(self):
+        self._cmd('azdev linter --include-whl-extensions {}'.format(self.extension_name))
+
+    def build(self):
+        pass
+
+
+def find_modified_files_against_master_branch():
+    """
+    Find modified files from src/ only.
+    A: Added, C: Copied, M: Modified, R: Renamed, T: File type changed.
+    Deleted files don't count in diff.
+    """
+    ado_pr_target_branch = 'origin/' + os.environ.get('ADO_PULL_REQUEST_TARGET_BRANCH')
+
+    separator_line()
+    print('pull request target branch:', ado_pr_target_branch)
+
+    cmd = 'git --no-pager diff --name-only --diff-filter=ACMRT {} -- src/'.format(ado_pr_target_branch)
+    files = check_output(cmd.split()).decode('utf-8').split('\n')
+    files = [f for f in files if len(f) > 0]
+
+    if files:
+        separator_line()
+        for f in files:
+            print(f)
+
+    return files
+
+
+def contain_index_json(files):
+    return 'src/index.json' in files
+
+
+def contain_extension_code(files):
+    with open('src/index.json', 'r') as fd:
+        current_extensions = json.loads(fd.read()).get("extensions")
+
+    extensions_home_prefixes = set('src/{}'.format(name) for name in current_extensions)
+
+    for file in files:
+        if any([file.startswith(prefix) for prefix in extensions_home_prefixes]):
+            return True
+
+    return False
+
+
+def linter_on_external_extension(index_json):
+    """
+    Check if the modified metadata items in index.json refer to the extension in repo.
+    If not, az extension linter on wheel. Otherwise skip it.
+    """
+
+    public_extensions = json.loads(check_output('az extension list-available -d', shell=True))
+
+    with open(index_json, 'r') as fd:
+        current_extensions = json.loads(fd.read()).get("extensions")
+
+    for name in current_extensions:
+        modified_entries = [entry for entry in current_extensions[name] if entry not in public_extensions.get(name, [])]
```

Comment:
Does it mean all extension which are deprecated would in `modified_entries`?

Reply:
with `-d`, the public_extensions will be populated with the exact content of raw index.json.